packages/ui/src/components/va-tree-view/VaTreeCategory/VaTreeCategory.vue

```diff
@@ -42,50 +42,57 @@
 </template>
 
 <script lang="ts">
-import { provide, inject, watch, ComponentPublicInstance, ref } from 'vue'
-import { Options, Vue, mixins, prop, setup } from 'vue-class-component'
+import { defineComponent, provide, inject, ref, watch, nextTick, PropType, ComponentPublicInstance } from 'vue'
+import { useColor } from '../../../composables/useColor'
 
-import ColorMixin from '../../../services/color-config/ColorMixin'
-import SquareWithIcon from '../SquareWithIcon'
-import VaIcon from '../../va-icon'
+// Components
+import SquareWithIcon from '../SquareWithIcon/SquareWithIcon.vue'
+import VaIcon from '../../va-icon/VaIcon.vue'
 import VaTreeNode from '../VaTreeNode/VaTreeNode.vue'
 
-class TreeCategoryProps {
-  label = prop<string | number>({ default: '', type: [String, Number] })
-  isOpen = prop<boolean>(Boolean)
-  icon = prop<string>({ default: '', type: String })
-  color = prop<string>({ type: String, default: 'primary' })
-}
-
-const TreeCategoryPropsMixin = Vue.with(TreeCategoryProps)
-
-@Options({
+const VaTreeCategory = defineComponent({
   name: 'VaTreeCategory',
-  components: { VaIcon, SquareWithIcon },
-})
-export default class VaTreeCategory extends mixins(
-  ColorMixin,
-  TreeCategoryPropsMixin,
-) {
-  isOpenCached: boolean | undefined = false
-
-  setupContext = setup(() => {
-    const nodes = ref<(VaTreeCategory | VaTreeNode)[]>([])
-
-    const onChildMounted = (node: VaTreeCategory | VaTreeNode) => {
+  components: {
+    SquareWithIcon,
+    VaIcon,
+  },
+  props: {
+    label: {
+      type: [String, Number] as PropType<string | number>,
+      default: '',
+    },
+    isOpen: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    icon: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    color: {
+      type: String as PropType<string>,
+      default: 'primary',
+    },
+  },
+  setup: (props) => {
+    const nodes: any = ref<(typeof VaTreeCategory | typeof VaTreeNode)[]>([])
+    const isOpenCached = ref<boolean | undefined>(false)
+
+    const onChildMounted = (node: typeof VaTreeCategory | typeof VaTreeNode) => {
       nodes.value.push(node)
     }
 
-    const onChildUnmounted = (removableNode: VaTreeCategory | VaTreeNode) => {
-      nodes.value = nodes.value.filter((node: VaTreeCategory | VaTreeNode) => node !== removableNode)
+    const onChildUnmounted = (removableNode: typeof VaTreeCategory | typeof VaTreeNode) => {
+      nodes.value = nodes.value.filter((node: typeof VaTreeCategory | typeof VaTreeNode) => node !== removableNode)
     }
 
-    const treeCategory = {
+    const treeCategory: any = {
       onChildMounted,
       onChildUnmounted,
     }
 
     provide('treeCategory', treeCategory)
```

Comment:
Overall we could also expose this symbol in lib, so it could be used outside of vuestic if someone wants it that hard.
But such usage would be very explicit.